pkgs/by-name/st/stackit-cli/package.nix

```
@@ -0,0 +1,84 @@
+{ lib
+, buildGoModule
+, fetchFromGitHub
+, installShellFiles
+, makeWrapper
+, less
+, xdg-utils
+, testers
+, runCommand
+, stackit-cli
+}:
+
+buildGoModule rec {
+  pname = "stackit-cli";
+  version = "0.1.0-apt-prerelease.3";
+
+  src = fetchFromGitHub {
+    owner = "stackitcloud";
+    repo = "stackit-cli";
+    rev = "v${version}";
+    hash = "sha256-WpIGWUhqjNjOnaK5pTKYgXjCasbfzSrjcnqcMTM9sBo=";
+  };
+
+  vendorHash = "sha256-nhOhS8fB48gOnlsUPlUprdp3iNeDO9AMFymCa7zsVSY=";
+
+  subPackages = [ "." ];
+
+  CGO_ENABLED = 0;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+  ];
+
+  nativeBuildInputs = [ installShellFiles makeWrapper ];
+
+  preCheck = ''
+    export HOME=$TMPDIR # needed because the binary always creates a dir & config file
+  '';
+
+  postInstall = ''
+    export HOME=$TMPDIR # needed because the binary always creates a dir & config file
+    mv $out/bin/{${pname},stackit} # rename the binary
+
+    installShellCompletion --cmd stackit --bash <($out/bin/stackit completion bash)
+    installShellCompletion --cmd stackit --zsh <($out/bin/stackit completion zsh)
+    installShellCompletion --cmd stackit --fish <($out/bin/stackit completion fish)
+    # Use this instead, once https://github.com/stackitcloud/stackit-cli/issues/153 is fixed:
+    # installShellCompletion --cmd stackit \
+    #   --bash <($out/bin/stackit completion bash) \
+    #   --zsh  <($out/bin/stackit completion zsh)  \
+    #   --fish <($out/bin/stackit completion fish)
+  '';
+
+  postFixup = ''
+    wrapProgram $out/bin/stackit \
+      --prefix PATH : ${lib.makeBinPath [ less xdg-utils ]}
```

```suggestion
      --suffix PATH : ${lib.makeBinPath [ less xdg-utils ]}
```
xdg-utils should be suffixed otherwise patches to xdg-open don't apply.

---

Is it fine that `less` is then also being suffixed?

---

Shouldn't hurt and make things easier